scripts/install_avalanchego_release.sh

```diff
@@ -27,20 +27,64 @@ if [[ ${GOOS} == "darwin" ]]; then
   AVAGO_DOWNLOAD_PATH=${BASEDIR}/avalanchego-macos-${VERSION}.zip
 fi
 
-AVALANCHEGO_BUILD_PATH=${AVALANCHEGO_BUILD_PATH:-${BASEDIR}/avalanchego-${VERSION}}
+# TODO: Please read:
+# not sure why the version isn't included in CI currently... open to suggestions on how to handle this better
+AVALANCHEGO_BUILD_PATH=${AVALANCHEGO_BUILD_PATH-${VERSION}:-${BASEDIR}/avalanchego-${VERSION}}
 mkdir -p $AVALANCHEGO_BUILD_PATH
 
 if [[ ! -f ${AVAGO_DOWNLOAD_PATH} ]]; then
   echo "downloading avalanchego ${VERSION} at ${AVAGO_DOWNLOAD_URL} to ${AVAGO_DOWNLOAD_PATH}"
-  curl -L ${AVAGO_DOWNLOAD_URL} -o ${AVAGO_DOWNLOAD_PATH}
+
+  # test if the tarball is already available for download
+  if curl -s --head --request GET ${AVAGO_DOWNLOAD_URL} | grep "302" > /dev/null; then
+    curl -L ${AVAGO_DOWNLOAD_URL} -o ${AVAGO_DOWNLOAD_PATH}
+  else
+    GIT_CLONE_URL=https://github.com/ava-labs/avalanchego.git
+    GIT_CLONE_PATH=${BASEDIR}/avalanchego-git/
+    mkdir -p $GIT_CLONE_PATH
+
+    WORKDIR=$PWD
+    cd $GIT_CLONE_PATH
+
+    # if the git repo already exists, fetch, otherwise clone
+    if [[ -d .git ]]; then
+      git fetch
+    else
+      git clone ${GIT_CLONE_URL} .
```

Comment:
should we make this a shallow clone of the repository and avoid fetching the entire repository if the dir exists?